### ISSUE TITLE
Maximum call stack size exceeded when use multiple assignment

### ISSUE THREAD

**Issue by rg-golubkov** [microsoft/pyright#508]
**Describe the bug**
When instances of a container class is used in multiple assignment (order is important) pyright raises error.
`RangeError: Maximum call stack size exceeded
`

**To Reproduce**
An example of not real code
``` python
from dataclasses import dataclass
from typing import Optional, TypeVar

T = TypeVar('T')

@dataclass
class Node:
    key: T
    next: Optional['Node'] = None

node = Node(0)
node.next, node = None, None # Fails here
```

**Expected behavior**
There should be no error issued.

**Screenshots or Code**
I got this error when writing a linked list implementation
![image](https://user-images.githubusercontent.com/33006675/74260071-f9d8c980-4d09-11ea-900c-0f7e0b7d134f.png)

**VS Code extension or command-line**
VS code extension (v1.1.21)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.22, which I just published.
